### PR TITLE
Update Jenkins plugin to new routing

### PR DESCRIPTION
**Pull request by timja**
## Hey, I just made a Pull Request!

See https://github.com/spotify/backstage/issues/2200

<details>

<summary>Screenshot diff</summary>

image

</details>

TODO:

- [x] Last build widget
- [x] Review README of plugin

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [x] Regression tests added for bug fixes

### CODE REVIEW

File: docs/features/software-catalog/well-known-annotations.md [hunk 1]
```diff
@@ -51,13 +51,13 @@ The value of this annotation is a location reference string (see above). If this
 annotation is specified, it is expected to point to a repository that the
 TechDocs system can read and generate docs from.
 
-### backstage.io/jenkins-github-folder
+### jenkins.io/github-folder
```
Comment (freben) [inline docs/features/software-catalog/well-known-annotations.md:54]: <3

File: plugins/jenkins/src/components/BuildWithStepsPage/BuildWithStepsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  const [{ value }] = useBuildWithSteps(buildPath);
+
+  return (
+    <div className={classes.root}>
+      <Breadcrumbs aria-label="breadcrumb">
+        <Link to="../../..">Jobs</Link>
+        <Typography>Run</Typography>
+      </Breadcrumbs>
+      <TableContainer component={Paper} className={classes.table}>
+        <Table>
```
Comment (stefanalund) [inline plugins/jenkins/src/components/BuildWithStepsPage/BuildWithStepsPage.tsx:70]: Out of curiosity, how come you don't use the Backstage Table component for this?

https://backstage.io/storybook/?path=/story/table--default-table

Comment (timja) [reply plugins/jenkins/src/components/BuildWithStepsPage/BuildWithStepsPage.tsx:70]: The code is a 100% copy paste from the github actions plugin, only where the data comes from is changed

Comment (timja) [reply plugins/jenkins/src/components/BuildWithStepsPage/BuildWithStepsPage.tsx:70]: This looks like a different component to that really.

The jenkins build list does use that component but it doesn't fit for this which is just displaying details.

tbh table probably isn't the best fix for it, but I was trying to make the plugins look similar.

**Comment by timja** [comment]
@freben @Rugvip any chance of a review please?

**Review by shmidt-i — APPROVED** [review]
All good, thanks!

File: docs/features/software-catalog/well-known-annotations.md [hunk 3]
```diff
@@ -51,13 +51,13 @@ The value of this annotation is a location reference string (see above). If this
...
 
 ```yaml
 # Example:
 metadata:
   annotations:
-    backstage.io/jenkins-github-folder: folder-name/job-name
+    jenkins.io/github-folder: folder-name/job-name
```
Comment (Rugvip) [inline docs/features/software-catalog/well-known-annotations.md:60]: 👍

**Review by Rugvip — APPROVED** [review]
Indeed, thanks! 👍

**Pull request merged**